File: Library/DataBinding/src/main/java/zhaoyun/techstack/android/databinding/MainActivity.java
package zhaoyun.techstack.android.databinding;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import zhaoyun.techstack.android.databinding.databinding.ActivityMainBinding;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        ViewModel viewModel = new ViewModel();
        binding.setViewModel(viewModel);
    }
}

File: Topic/Arch/Mvvm/src/main/java/zhaoyun/techstack/android/mvvm/MainActivity.kt
package zhaoyun.techstack.android.mvvm

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.ViewModelProviders
import kotlinx.android.synthetic.main.activity_main.*
import zhaoyun.techstack.mvvm.R

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val animeViewModel: AnimeViewModel = ViewModelProviders.of(this).get(AnimeViewModel::class.java)

        anime_view.apply {
            setOnClickListener {
                animeViewModel.fetchNewSeason()
            }
            setData(animeViewModel.animeList)
        }
    }
}
